### ISSUE TITLE
Inconsistent parameter reloading

### ISSUE THREAD

**Issue by pblouw** [nengo/nengo_dl#9]
Reloading model parameters can introduce unexpected indeterminacy to simulations. Minimal example:

```python
import nengo
import nengo.spa as spa
import nengo_dl
import numpy as np
import tensorflow as tf

seed = 98
dims = 32

vocab = spa.Vocabulary(dimensions=dims)
vocab.parse('TRACE')
vocab.parse('CUE')
vocab.add('OUTPUT', vocab.parse('TRACE*~CUE').v)

with nengo.Network(seed=seed) as net:
    net.config[nengo.Ensemble].neuron_type = nengo.RectifiedLinear()
    net.config[nengo.Connection].synapse = None
    
    trace_inp = nengo.Node(vocab['TRACE'].v)
    cue_inp = nengo.Node(vocab['CUE'].v)
    
    extractor = nengo.networks.CircularConvolution(5, dims, invert_b=True)

    nengo.Connection(trace_inp, extractor.input_a)
    nengo.Connection(cue_inp, extractor.input_b)

    out = nengo.Probe(extractor.output)
    
inp_array = vocab['TRACE'].v 
inp_array = inp_array[None, None, :]
cue_array = vocab['CUE'].v
cue_array = cue_array[None, None, :]
out_array = vocab['OUTPUT'].v
out_array = out_array[None, None, :]
    
inputs = {trace_inp: inp_array, cue_inp: cue_array}
outputs = {out: out_array}
        
with nengo_dl.Simulator(net, seed=seed) as sim1:
    optimizer = tf.train.RMSPropOptimizer(1e-3)    
    print('loss pre-training ', sim1.loss(inputs, outputs, 'mse'))
    sim1.train(inputs, outputs, optimizer, n_epochs=5, objective='mse')
    print('loss post-training ', sim1.loss(inputs, outputs, 'mse'))
    sim1.save_params('./example-params')
    
with nengo_dl.Simulator(net, seed=seed) as sim2:
    sim2.load_params('./example-params')
    print('loss post-reloading', sim2.loss(inputs, outputs, 'mse'))
```

The loss computed after reloading the saved model parameters will not be equivalent to the loss computed prior to saving these parameters.
